### DISCUSSION TITLE
Is there a reason to use Documents vs Vertexes?

### DISCUSSION THREAD

**Discussion by adubkov** [ArcadeData/arcadedb#896] [Q&A]
It seems that Vertexes can have nested fields, just like Documents, in addition to IN and OUT edges. Is there a reason to use Documents at all?

**Comment by lvca** [comment]
If you're not connecting documents and you just need inside vertices as embedded, there is no much reason to use Documents. After all, a Vertex is a Document with the chance to be connected to other vertices.

**Comment by adubkov** [comment]
I also found couple of more differences in OrientDB [documentation](https://orientdb.com/docs/2.2.x/Choosing-between-Graph-or-Document-API.html#document-api). Not sure if same applies to ArcadeDB though:

>- Relationships are only **mono-directional**. If you need bidirectional relationships, it is your responsibility to maintain both LINKs.
>- A Document is an atomic unit, while with Graphs, the relationships are modeled through In and Out properties. For this reason, Graph operations must be done within transactions. In contrast, when you create a relationship between documents with a LINK, the targeted linked document is not involved in this operation. This results in **better Multi-Threaded support**, especially with insert, delete and update operations.

**Reply by lvca** [reply]
they are both valid with ArcadeDB too, but the difference in overhead between documents and edges in OrientDB is much bigger than in ArcadeDB which manages vertices in a much more efficient way. Also, ArcadeDB supported unidirectional edges that are, in practice, links in documents with OrientDB.

Also, with ArcadeDB there is no penalty with multi-threads and using vertices like it as with OrientDB.
